### DISCUSSION TITLE
Delete a row but the whole data deleted

### DISCUSSION THREAD

**Discussion by letscodetpl** [serverpod/serverpod#2374] [Q&A] [answered]
Hi All,

I have an issue where I want to delete a row in DB but it ends up delete entire data in DB. Not sure what I did wrong.
I use Listview.builder to return a custom tile which pass in an object variables and 1 void Function()? onPressed.
Below is my code. Thanks for your help.

My Endpoint
![image](https://github.com/serverpod/serverpod/assets/171565733/f75105ff-21ef-4649-ad5e-6af5bed30ed3)

My custom tile
![image](https://github.com/serverpod/serverpod/assets/171565733/214a0205-caf4-458c-9076-3450f577ad19)

My UI page
![image](https://github.com/serverpod/serverpod/assets/171565733/95fc0eef-34fd-47b9-88b3-3ad164170ebc)

![image](https://github.com/serverpod/serverpod/assets/171565733/1ab3b805-eb82-4d39-a5ff-9f34862a532b)

**Comment by SandPod** [comment, marked as answer]
Hmm, shouldn't your `onPressed` be a callback?

So change `onPressed: deleteVisitor(_visitors![index].id!),` to `onPressed: () => deleteVisitor(_visitors![index].id!)`.

If you have not already done this, a good sanity check would be to add a debug print to the `deleteAVisitor` endpoint on the server to ensure that you are not accidentally calling that endpoint with all the visitor IDs from another place in the app.

Do you have the possibility to inspect the database query being performed using the insights tool? https://docs.serverpod.dev/insights
The tool allows you to enable logging for endpoint calls and DB queries.

That way we could inspect what the generated DB query is.

**Reply by letscodetpl** [reply]
thank youuu!
